File: lab2/lab_python_oop/circle.py
from lab_python_oop.geometric_figure import geometric_figure
from lab_python_oop.figure_color import figure_color
import math

class Circle(geometric_figure):
    figure_type = "Круг"
    def __init__(self, radius, color):
        self.radius = radius
        self.color = figure_color()
        self.color.color = color
    def get_area(self):
        return math.pi*self.radius*self.radius
    @classmethod
    def get_type(cls):
        return cls.figure_type
    def __repr__(self):
       return 'Тип: {}, Радиус: {}, Цвет: {}, Площадь: {}'.format(Circle.get_type(), self.radius, self.color.color, self.get_area())

File: lab3_rust/geometrics/figures/src/circle.rs
use crate::color::Color;
use std::fmt;
use crate::figure::Figure;

pub struct Circle{
    figure_type: String,
    radius: f64,
    color: Color
}

impl Circle{
    pub fn new(rad: f64, col: &Color) -> Circle{
        let c: Color = Color::new(col.get_color());
        Circle {figure_type: "circle".to_string(), radius: rad, color: c}
    }
    pub fn get_type(&self) -> String{
        self.figure_type.clone()
    }
}

impl Figure for Circle{
    fn get_area(&self) -> f64{
        std::f64::consts::PI*self.radius.powi(2)
    }
    fn repr(&self) -> String{
        format!("Тип: {}, Радиус: {}, Цвет: {}, Площадь: {}", self.get_type(), self.radius, self.color.get_color(), self.get_area())
    }
}

impl std::fmt::Display for Circle{
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{}", self.repr())
    }
}